robottelo/ui/computeresource.py

```diff
@@ -441,3 +442,23 @@ def set_profile_values(self, resource_name, profile_name, **kwargs):
             self, resource_type)
         resource_profile_form.set_values(**kwargs)
         resource_profile_form.submit()
+
+    def check_image_os(self, os_name):
+        """Check if the OS is present, if not create the required OS
+
+        :param os_name: OS name to check, and create
+        :return Created os
+        """
+        # Check if OS that image needs is present or no, If not create the OS
+        result = entities.OperatingSystem().search(query={
+            u'search': u'title="{0}"'.format(os_name)
+        })
+        os = entities.OperatingSystem().create()
+        if len(result) == 0 or (os_name != result[0].title):
+            os = entities.OperatingSystem(
+                id=os.id,
+                name=os_name.split(' ')[0],
+                major=os_name.split(' ')[1].split('.')[0],
+                minor=os_name.split(' ')[1].split('.')[1],
+            ).update(['name', 'major', 'minor'])
+        return os
```

Comment:
This will create a issue, that we are tackling right now, as if there will be a default os we will end up deleting at the end of the test and that could be problematic as we have not created it , so we should not delete it. The test will try to delete the used os.i.e Redhat 7.3. To avoid interfering with Default OS , we are creating new os, updating it (if no default os), deleting it. This way we will avoid deleting the Default os.

```
2017-07-24 10:12:46 - nailgun.client - DEBUG - Making HTTP DELETE request to https://qe-sat6-feature-rhel6/api/v2/operatingsystems/34 with options {'verify': Fal
se, 'auth': ('admin', 'changeme'), 'headers': {'content-type': 'application/json'}} and no data.
2017-07-24 10:12:47 - nailgun.client - WARNING - Received HTTP 422 response: {
  "error": {"id":34,"errors":{"base":["RedHat 7.3 is used by david-blasi.","RedHat 7.3 is used by petra-drossos.","RedHat 7.3 is
 used by qe-sat6-feature-rhel6","RedHat 7.3 is used by aimee-vietzke.host.centralci.eng.rdu2.redhat.com","RedHat 7.3 is used by ethan-molinari."]},"full_messages":["RedHat 7.3 is used by david-blasi.","RedHat 7.3 is used by petra-drossos.","RedHat 7.3 is
used by qe-sat6-feature-rhel6t.com","RedHat 7.3 is used by aimee-vietzke..com","RedHat 7.3 is used by ethan-molinari.com"]}
}

2017-07-24 10:12:47 - robottelo - DEBUG - Finished Test: RhevComputeResourceTestCase/test_positive_add_image_rhev_with_name
2017-07-24 10:12:47 - robottelo.test - DEBUG - Saving screenshot /tmp/robottelo/screenshots
```